src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoCondition } from '../../types/TodoCondition';
+
+type Props = {
+  todo: Todo,
+  todoCondition: TodoCondition,
+  onDeleteTodo?: (todoId: number) => void,
+  toggleTodo?: (curentTodo: Todo, status?: boolean | undefined) => void,
+  handleSubmitEditing?: (id: number, thisTitle: string) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  todoCondition,
+  onDeleteTodo = () => { },
+  toggleTodo = () => { },
+  handleSubmitEditing = () => { },
+}) => {
+  const { id, title, completed } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [thisTitle, setThisTitle] = useState(title);
```

Comment:
consider more readable naming
```suggestion
  const [editTitle, setEditTitle] = useState(title);
```
or updatedTitle, newTitle